src/main/java/io/github/lottetreg/httpserver/HTTPHeaders.java

```diff
@@ -1,33 +1,45 @@
 package io.github.lottetreg.httpserver;
 
+import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 import static java.util.stream.Collectors.toMap;
 
 public class HTTPHeaders {
-  private Map<String, String> headers;
+  private HashMap<String, String> headers;
   private String headerSeparator = ": ";
 
   HTTPHeaders(List<String> headersList) {
-    this.headers = headersList.stream().collect(toMap(
-            header -> getHeaderName(header),
-            header -> getHeaderValue(header)));
+    this.headers = new HashMap<>(headersList.stream().collect(toMap(
+        header -> getHeaderName(header),
+        header -> getHeaderValue(header))));
+  }
+
+  HTTPHeaders() {
+    this.headers = new HashMap<>();
+  }
+
+  public HashMap<String, String> getHeaders() {
+    return this.headers;
   }
 
   public String getHeader(String headerName) {
     return this.headers.get(headerName);
   }
 
-  public String getHeaderOrDefault(String headerName, String defaultValue) {
+  public String getHeader(String headerName, String defaultValue) {
     return this.headers.getOrDefault(headerName, defaultValue);
   }
 
+  public void addHeader(String headerName, String headerValue) {
+    this.headers.put(headerName, headerValue);
+  }
+
   private String getHeaderName(String header) {
     String headerName = splitHeader(header)[0];
 
-    if(headerName.isEmpty()) {
-      throw new IncorrectlyFormattedHeaderException(header);
+    if (headerName.isEmpty()) {
+      throw new IncorrectlyFormattedHeader(header);
```

Comment:
I think in most places you are checking for `isEmpty` you could easily swap for using optionals

Reply:
Ahh yes good call. I'm going to have to audit the older code and see where I can use Optionals instead. Thank you!